engine/access/rest/websockets/data_providers/factory_test.go

```diff
@@ -0,0 +1,132 @@
+	suite.Suite
+
+	ctx context.Context
+	ch  chan interface{}
+
+	accessApi      *accessmock.API
+	stateStreamApi *statestreammock.API
+
+	factory *DataProviderFactoryImpl
+}
+
+func TestDataProviderFactorySuite(t *testing.T) {
+	suite.Run(t, new(DataProviderFactorySuite))
+}
+
+// SetupTest sets up the initial context and dependencies for each test case.
+// It initializes the factory with mock instances and validates that it is created successfully.
+func (s *DataProviderFactorySuite) SetupTest() {
+	log := unittest.Logger()
+	s.stateStreamApi = statestreammock.NewAPI(s.T())
+	s.accessApi = accessmock.NewAPI(s.T())
+
+	s.ctx = context.Background()
+	s.ch = make(chan interface{})
+
+	s.factory = NewDataProviderFactory(log, s.stateStreamApi, s.accessApi)
+	s.Require().NotNil(s.factory)
+}
+
+// setupSubscription creates a mock subscription instance for testing purposes.
+// It configures the return value of the specified API call to the mock subscription.
+func (s *DataProviderFactorySuite) setupSubscription(apiCall *mock.Call) {
+	subscription := statestreammock.NewSubscription(s.T())
+	apiCall.Return(subscription).Once()
+}
+
+// TODO: add others topic to check when they will be implemented
+// TestSupportedTopics verifies that supported topics return a valid provider and no errors.
+// Each test case includes a topic and arguments for which a data provider should be created.
+func (s *DataProviderFactorySuite) TestSupportedTopics() {
+	// Define supported topics and check if each returns the correct provider without errors
+	testCases := []struct {
+		name               string
+		topic              string
+		arguments          map[string]string
+		setupSubscription  func()
+		assertExpectations func()
+	}{
+		{
+			name:      "block topic",
+			topic:     BlocksTopic,
+			arguments: map[string]string{"block_status": parser.Finalized},
+			setupSubscription: func() {
+				s.setupSubscription(s.accessApi.On("SubscribeBlocksFromLatest", mock.Anything, flow.BlockStatusFinalized))
+			},
+			assertExpectations: func() {
+				s.accessApi.AssertExpectations(s.T())
+			},
+		},
+		{
+			name:      "block headers topic",
+			topic:     BlockHeadersTopic,
+			arguments: map[string]string{"block_status": parser.Finalized},
+			setupSubscription: func() {
+				s.setupSubscription(s.accessApi.On("SubscribeBlockHeadersFromLatest", mock.Anything, flow.BlockStatusFinalized))
+			},
+			assertExpectations: func() {
+				s.accessApi.AssertExpectations(s.T())
+			},
+		},
+		{
+			name:      "block digests topic",
+			topic:     BlockDigestsTopic,
+			arguments: map[string]string{"block_status": parser.Finalized},
+			setupSubscription: func() {
+				s.setupSubscription(s.accessApi.On("SubscribeBlockDigestsFromLatest", mock.Anything, flow.BlockStatusFinalized))
+			},
+			assertExpectations: func() {
+				s.accessApi.AssertExpectations(s.T())
+			},
+		},
+	}
+
+	for _, test := range testCases {
+		s.Run(test.name, func() {
+			test.setupSubscription()
```

Comment:
nit: I guess we can run them in parallel

```suggestion
                        s.T().Parallel()
			test.setupSubscription()
```

Reply:
[5581550](https://github.com/onflow/flow-go/pull/6636/commits/558155049a830b31cf5700a158ba878abc346c1a)